src/internal/presentation/cli/state/game.go

```diff
@@ -57,43 +63,145 @@ func NewGame() (*Game, error) {
+		action.MoveDown:             {X: 0, Y: 1},
+		action.MoveLeft:             {X: -1, Y: 0},
+		action.MoveRight:            {X: 1, Y: 0},
+		action.MoveLeftUpperCorner:  {X: -1, Y: -1},
+		action.MoveRightUpperCorner: {X: 1, Y: -1},
+		action.MoveLefLowerCorner:   {X: -1, Y: 1},
+		action.MoveRightLowerCorner: {X: 1, Y: 1},
+	}
 	game.view.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
 		switch game.eventToAction(event) {
-		case action.MoveUp:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: 0, Y: -1})
-			return nil
-		case action.MoveDown:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: 0, Y: 1})
-			return nil
-		case action.MoveLeft:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: -1, Y: 0})
-			return nil
-		case action.MoveRight:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: 1, Y: 0})
-			return nil
-		case action.MoveLeftUpperCorner:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: -1, Y: -1})
-			return nil
-		case action.MoveRightUpperCorner:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: 1, Y: -1})
-			return nil
-		case action.MoveLefLowerCorner:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: -1, Y: 1})
-			return nil
-		case action.MoveRightLowerCorner:
-			game.player.Character.Shape.Move(primitives.Point2D[int]{X: 1, Y: 1})
-			return nil
+		case action.MoveUp,
+			action.MoveDown,
+			action.MoveLeft,
+			action.MoveRight,
+			action.MoveLeftUpperCorner,
+			action.MoveRightUpperCorner,
+			action.MoveLefLowerCorner,
+			action.MoveRightLowerCorner:
+			oldPosition := game.moveAndGetOldPosition(movement[game.eventToAction(event)])
+			if game.checkCollision(oldPosition) {
+				game.player.Character.Shape.Point = oldPosition
+			}
 		case action.Exit:
 			game.signal = signals.Stop
 			return nil
 		default:
 			return event
 		}
+		return nil
 	})
 
 	return &game, nil
 }
 
+func (g *Game) moveAndGetOldPosition(changingPosition primitives.Point2D[int]) primitives.Point2D[int] {
+	oldPosition := g.player.Character.Shape.Point
+	g.player.Character.Shape.Move(changingPosition)
+	return oldPosition
+}
+
+func (g *Game) checkCollision(oldPosition primitives.Point2D[int]) bool {
+	playerPosition := g.player.Character.Shape.Point
+
+	if playerPosition.X < 0 || playerPosition.Y < 0 {
+		return true
+	}
+	if playerPosition.X >= WidthLevel || playerPosition.Y >= HeightLevel {
+		return true
+	}
+	if g.checkCollisionInRoom() {
+		return true
+	}
+	if g.checkCollisionWithEnemy() {
+		return true
+	}
+	if inPassage, hasCollision := g.checkCollisionInPassage(oldPosition); inPassage {
+		if !hasCollision {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (g *Game) checkCollisionInRoom() bool {
+	playerPosition := g.player.Character.Shape.Point
+
+	for _, room := range g.level.Rooms {
+		if isPlayerInWall(playerPosition, room) {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
Странное название функции, т.к. разве только со стенками мы можем сталкиваться в комнате? Если нам надо проверить коллизию со стенкой, то так функцию и следует назвать:

```suggestion
func (g *Game) checkWithWall() bool {
	playerPosition := g.player.Character.Shape.Point

	for _, room := range g.level.Rooms {
		if isPlayerInWall(playerPosition, room) {
			return true
		}
	}

	return false
}
```